bukkit/src/main/java/org/wensheng/juicyraspberrypie/RemoteSession.java

```diff
@@ -429,11 +448,73 @@ private void handleEntityCommand(String c, String[] args, boolean entityIsPlayer
                 }
                 Location loc = entity.getLocation();
                 entity.teleport(parseLocation(x, y, z, loc.getPitch(), loc.getYaw()));
+                send("ok");
+                break;
+            }
+            case "enableControl": {
+                if (entity instanceof Mob mob) {
+                    if(!mob.hasAI()) {
+                        mob.getPersistentDataContainer().set(new NamespacedKey(plugin, "noAI"), PersistentDataType.BOOLEAN, true);
+                        mob.setAI(true);
+                    }
+                    final MobGoals mobGoals = Bukkit.getMobGoals();
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noMoveAI", GoalType.MOVE));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noTargetAI", GoalType.TARGET));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noJumpAI", GoalType.JUMP));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noLookAI", GoalType.LOOK));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noUnknownBehaviorAI", GoalType.UNKNOWN_BEHAVIOR));
+                }
+                send("ok");
+                break;
+            }
+            case "disableControl": {
+                if (entity instanceof Mob mob) {
+                    final MobGoals mobGoals = Bukkit.getMobGoals();
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noMoveAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noTargetAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noJumpAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noLookAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noUnknownBehaviorAI")));
+                    final PersistentDataContainer persistentDataContainer = mob.getPersistentDataContainer();
+                    final boolean noAI = persistentDataContainer.has(new NamespacedKey(plugin, "noAI"), PersistentDataType.BOOLEAN);
+                    if (noAI) {
+                        persistentDataContainer.remove(new NamespacedKey(plugin, "noAI"));
+                        mob.setAI(false);
+                    }
+                }
+                send("ok");
+                break;
+            }
+            case "walkTo": {
+                if (entity instanceof Mob mob) {
+                    String x = args[0], y = args[1], z = args[2];
+                    if (args.length > 3) {
+                        x = args[1];
+                        y = args[2];
+                        z = args[3];
+                    }
+                    final Location loc = parseLocation(x, y, z);

```

Comment:
This fragment already was duplicated, now even more. But I'm afraid for that the whole switch statement would have to be broken up first...